sdk/eventgrid/azure-eventgrid/tests/asynctestcase.py

```diff
@@ -0,0 +1,40 @@
+
+# coding: utf-8
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import asyncio
+import os
+import functools
+from azure_devtools.scenario_tests.utilities import trim_kwargs_from_test_function
+from azure.core.credentials import AccessToken
+from eventgrid_preparer import EventGridTest
+
+
+class AsyncFakeTokenCredential(object):
+    """Protocol for classes able to provide OAuth tokens.
+    :param str scopes: Lets you specify the type of access needed.
+    """
+    def __init__(self):
+        self.token = AccessToken("YOU SHALL NOT PASS", 0)
+
+    async def get_token(self, *args):
+        return self.token
+
+
+class AsyncEventGridTest(EventGridTest):
+
+    def generate_oauth_token(self):
```

Comment:
You shouldn't need that, there is everything you need in devtools to care care of that for free